pages/instructorAssessmentQuestionManualGrading/instructorAssessmentQuestionManualGrading.ejs

```diff
@@ -2,18 +2,272 @@
+                  },
+                  {
+                      field: 'last_grader',
+                      title: 'Graded by',
+                      filterControl: 'select',
+                      searchFormatter: false,
+                      formatter: (value, row) => row.last_grader ? row.last_grader_name : '&mdash;',
+                  },
+              ]],
+          });
       });
+
+      function ajaxSubmit(e) {
+          e.preventDefault();
+          const submitter = e.submitter || e.originalEvent?.submitter;
+          if (submitter) {
+              // Obtain information from submit button that was used
+              $(e.target).find('input[name=batch_action_data]')
+                  .val(JSON.stringify($(submitter).data('batch-action') || {}));
+          }
+          $.post($(e.target).attr('action'), $(e.target).serialize(), function(data) {
+              $('#grading-table').bootstrapTable('refresh');
+          }, 'json').fail(function(data) {
+              // TODO Better user notification of update failure
+          });
+      }
+
+      function gradingTagDropdown() {
+          const course_staff = JSON.parse('<%- JSON.stringify(course_staff) %>') || [];
+          const dropdown = $('<div class="dropdown ml-1">');
+          const button = $('<button class="btn btn-secondary dropdown-toggle grading-tag-button" data-toggle="dropdown" name="status" disabled>').appendTo(dropdown);
+          button.text(' Tag for grading').prepend($('<i class="fas fa-tags">'));
+
+          const menu = $('<div class="dropdown-menu dropdown-menu-right">').appendTo(dropdown);
+          menu.append('<div class="dropdown-header">Tag as requiring grading</div>');
+
+          course_staff.forEach((grader) => {
+              const item_grader = $('<button class="dropdown-item" type="submit">').appendTo(menu);
+              item_grader.attr('data-batch-action', JSON.stringify({
+                  requires_manual_grading: true,
+                  assigned_grader: grader.user_id,
+              }));
+              item_grader.text(` Assign to: ${grader.name || ''} (${grader.uid})`);
+              item_grader.prepend('<i class="fas fa-user-tag"></i>');
+          });
+
+          const item_no_grader = $('<button class="dropdown-item" type="submit">').appendTo(menu);
+          item_no_grader.attr('data-batch-action', JSON.stringify({
+              requires_manual_grading: true,
+              assigned_grader: null,
+          }));
+          item_no_grader.text(` No grader assigned`).prepend('<i class="fas fa-tag"></i>');
+
+          menu.append('<div class="dropdown-divider">');
+
+          const item_graded = $('<button class="dropdown-item" type="submit">').appendTo(menu);
+          item_graded.attr('data-batch-action', JSON.stringify({
+              requires_manual_grading: false,
+          }));
+          item_graded.text(` Tag as graded`).prepend('<i class="fas fa-check-square"></i>');
```

Comment:
There is: tag as requiring grading with no grader assigned. Do you mean a tag that keeps the current assigned grader?

Reply:
Ahh, that makes sense now that you point it out. Maybe my problem is that there weren't any staff listed adjacent to "No grader assigned", so I interpreted that as a current status instead of an action?

Thinking out loud here, but maybe we could separate "update grading needed" and "assign graders" to make this clearer? I'm thinking something like the following (copy/icons all subject to change, of course):

<img width="241" alt="Screen Shot 2022-03-16 at 09 17 25" src="https://user-images.githubusercontent.com/1476544/158637129-e8a1185c-f57c-423c-bb76-e1a6c82813b4.png">

Reply:
Typically if you are assigning grading to someone you are also tagging it for grading (doesn't make sense to assign a grader without also tagging it for grader, from an instructor's perspective). While we can still maintain the actions above and just reshuffle the buttons, that could be confusing for instructors.

One option is to add a header to the second section, similar to the first section, to help clarify that they are different options?

Reply:
Ah, yes - we could still have the "Assign grader" actions _also_ tag as ungraded?

I think the confusing thing for me is that there are two pieces of state ("is graded" and "who is assigned") that are kind of independent and kind of not, and we have one dropdown where different actions control different subsets of that state 😅 maybe I'm overcomplicating this...

Reply:
Updated to match behaviour above.